### PR TITLE
docs: update CLI Usage page in the Sphinx documentation

### LINKED ISSUE #239: Add the content for the CLI options of Macaron in the Sphinx
**Description**
The CLI Options page has been created as a place holder (see [here](https://github.com/oracle-samples/macaron/blob/staging/docs/source/pages/cli_options.rst)). 
This page is for documenting the CLI options we have for Macaron. It contains the instructions on how to use the Macaron CLI. 

Blocking issue: #198

### PR DESCRIPTION
**Pull request by tromai**
Closes #239 .

### CODE REVIEW

File: docs/source/pages/cli_usage/action_dump_defaults.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Dumps the ``defaults.ini`` file to the output directory.
+
+-----
+Usage
+-----
+
+.. code-block:: shell
+
+    usage: macaron dump-defaults [-h]
```
Comment (behnazh-w) [inline docs/source/pages/cli_usage/action_dump_defaults.rst:17]: This should be `./run_macaron.sh dump-defaults`.

File: docs/source/pages/installation.rst [hunk 2]
```diff
@@ -36,12 +38,14 @@ To verify your setup, go to the directory containing the downloaded ``run_macaro
 
 .. code-block:: shell
 
-  ./run_macaron.sh --help
+  ./run_macaron.sh macaron --help
```
Comment (behnazh-w) [inline docs/source/pages/installation.rst:41]: No need to add `macaron` (similar in other places):
```suggestion
  ./run_macaron.sh --help
```

File: docs/source/pages/cli_usage/action_dump_defaults.rst (hunk 1 above)
Comment (tromai) [reply docs/source/pages/cli_usage/action_dump_defaults.rst:17]: I wonder if we want to put the CLI Options separately from whatever the user run it with?
So in this case we only need to document the flags and action commands (analyze, verify-policy, dump-defaults) and instruct that the user could provide it directly to:
 - `macaron` if installed as a Python package
 - `./run_macaron.sh` if run as a Docker image
Or should we only give instructions on using Macaron as a Docker image for this release?

Comment (tromai) [reply docs/source/pages/cli_usage/action_dump_defaults.rst:17]: I wonder if we want to put the CLI Options separately from whatever the user run it with?
So in this case we only need to document the flags and action commands (analyze, verify-policy, dump-defaults) and instruct that the user could provide it directly to:
 - `macaron` if installed as a Python package
 - `./run_macaron.sh` if run as a Docker image
Or should we only give instructions on using Macaron as a Docker image for this release?

Comment (behnazh-w) [reply docs/source/pages/cli_usage/action_dump_defaults.rst:17]: For now we assume users run Macaron as a Docker container. So let's just add `./run_macaron.sh` in the main CLI documentation for now and add `macaron` Python instructions just for contributors later.

I have created a ticket for that: https://github.com/oracle-samples/macaron/issues/272

File: docs/source/pages/installation.rst (hunk 2 above)
Comment (tromai) [reply docs/source/pages/installation.rst]: Addressed in https://github.com/oracle-samples/macaron/pull/269/commits/fc12961486572c6e91d62fa014813a49bf96df07

File: docs/source/pages/cli_usage/action_dump_defaults.rst (hunk 1 above)
Comment (tromai) [reply docs/source/pages/cli_usage/action_dump_defaults.rst]: Addressed in https://github.com/oracle-samples/macaron/pull/269/commits/fc12961486572c6e91d62fa014813a49bf96df07

**Comment by tromai** [comment]
I will rebase on staging to get the latest updates of the CLI options (from https://github.com/oracle-samples/macaron/pull/270).

File: docs/source/pages/cli_usage/action_analyze.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
+=======
+Analyze
+=======
+
+-----------
+Description
+-----------
+
+Analyze a Github public repository (and potentially the repositories of it dependencies) to determine its SLSA posture following the specification of `SLSA v0.1 <https://slsa.dev/spec/v0.1/>`_.
```
Comment (behnazh-w) [inline docs/source/pages/cli_usage/action_analyze.rst:9]:
```suggestion
Analyze a public GitHub repository (and optionally the repositories of its dependencies) to determine its SLSA posture. We currently support `SLSA v0.1 <https://slsa.dev/spec/v0.1/>`_. The support for `SLSA v1.0 <https://slsa.dev/spec/v1.0/>`_ will be added in future.
```

File: docs/source/pages/cli_usage/action_analyze.rst [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+.. option:: -g TEMPLATE_PATH, --template-path TEMPLATE_PATH
+
+    The path to the Jinja2 html template (please make sure to use .html or .j2 extensions).
+
+-----------
+Environment
+-----------
+
+``GITHUB_TOKEN`` – The GitHub personal access token needed for Macaron to run the analysis. For more information on how to obtain a token for Macaron, please see instructions in :ref:`Prepare GitHub access token <prepare-github-token>`.
```
Comment (behnazh-w) [inline docs/source/pages/cli_usage/action_analyze.rst:67]:
```suggestion
``GITHUB_TOKEN`` – The GitHub personal access token is needed for to run the analysis. For more information on how to obtain a GitHub token, see instructions in :ref:`Prepare GitHub access token <prepare-github-token>`.
```

File: docs/source/pages/cli_usage/action_dump_defaults.rst [hunk 5]
```diff
@@ -0,0 +1,25 @@
+=============
+Dump Defaults
+=============
+
+-----------
+Description
+-----------
+
+Dumps the ``defaults.ini`` file to the output directory.
```
Comment (behnazh-w) [inline docs/source/pages/cli_usage/action_dump_defaults.rst:9]:
```suggestion
Dumps the ``defaults.ini`` configuration file used by Macaron to the output directory. You can make changes to this configuration file and pass it to Macaron using the `--defaults-path` option. See <reference-to-analyze-action>` for more information.
```

File: docs/source/pages/cli_usage/index.rst [hunk 6]
```diff
@@ -0,0 +1,61 @@
+.. Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+.. Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+.. _cli-usage:
+
+==================
+Command Line Usage
+==================
+
+You could use the bash script ``run_macaron.sh`` we provided to run Macaron as a Docker image (For more information on how to get this script, please see :ref:`Download Macaron <download-macaron>`).
```
Comment (behnazh-w) [inline docs/source/pages/cli_usage/index.rst:10]:
```suggestion
Use the bash script ``run_macaron.sh`` to run Macaron as a Docker container (for more information on how to get this script, please see :ref:`Download Macaron <download-macaron>`).
```

File: docs/source/pages/installation.rst [hunk 7]
```diff
@@ -41,7 +43,9 @@ To verify your setup, go to the directory containing the downloaded ``run_macaro
 
 .. note:: In the first execution, this script will download the Macaron Docker image from ``ghcr.io/oracle-samples/macaron`` which can take some time. However, the next time you run it, the docker image available on your local host will be used.
 
-.. note:: By default, ``latest`` is used as the tag for the downloaded image. You could specify the tag you want to run by assigning the environment variable ``MACARON_IMAGE_TAG``. For example to run Macaron v0.1: ``MACARON_IMAGE_TAG=0.1 && ./run_macaron.sh --help``
+.. note:: By default, ``latest`` is used as the tag for the downloaded image. You could specify the tag you want to run by assigning the environment variable ``MACARON_IMAGE_TAG``. For example to run Macaron v0.1: ``MACARON_IMAGE_TAG=0.1 && ./run_macaron.sh macaron --help``
```
Comment (behnazh-w) [inline docs/source/pages/installation.rst:46]:
```suggestion
.. note:: By default, ``latest`` is used as the tag for the downloaded image. You can choose a specific tag by assigning the environment variable ``MACARON_IMAGE_TAG``. For example to run Macaron v0.1.0 run: ``MACARON_IMAGE_TAG=v0.1.0 && ./run_macaron.sh --help``

Not that the image tag starts with `v`.
```

File: docs/source/pages/using.rst [hunk 8]
```diff
@@ -19,29 +19,29 @@ To run Macaron on a Github public repository, we use the following command:
 
 .. code-block:: shell
 
-  ./run_macaron.sh analyze -rp <repo_path>
+  ./run_macaron.sh macaron analyze -rp <repo_path>
```
Comment (behnazh-w) [inline docs/source/pages/using.rst:22]: This file still contains `./run_macaron.sh macaron` command. Please fix these.

File: docs/source/pages/installation.rst (hunk 7 above)
Comment (tromai) [reply docs/source/pages/installation.rst:46]: By ``Not that the image tag starts with `v`.`` do you mean ``Note that the image tag starts with `-v`.`` Do you think we should use a different `.. note::` derivative for it or we could put it directly after the previous `.. note::`?

Comment (behnazh-w) [reply docs/source/pages/installation.rst:46]: > By `` Not that the image tag starts with `v`. ``

Yes, sorry :sweat_smile: That was just for your information. No need to add it as a note.

Comment (tromai) [reply docs/source/pages/installation.rst:46]: Oh I see. No worries :grin: Thank you.

File: docs/source/pages/using.rst (hunk 8 above)
Comment (tromai) [reply docs/source/pages/using.rst]: Addressed in https://github.com/oracle-samples/macaron/pull/269/commits/602d68376381eb591484d8df2ecaee0e26dd0e7e

File: docs/source/pages/installation.rst (hunk 7 above)
Comment (tromai) [reply docs/source/pages/installation.rst]: Addressed in https://github.com/oracle-samples/macaron/pull/269/commits/602d68376381eb591484d8df2ecaee0e26dd0e7e

File: docs/source/pages/cli_usage/index.rst (hunk 6 above)
Comment (tromai) [reply docs/source/pages/cli_usage/index.rst]: Addressed in https://github.com/oracle-samples/macaron/pull/269/commits/602d68376381eb591484d8df2ecaee0e26dd0e7e

File: docs/source/pages/cli_usage/action_dump_defaults.rst (hunk 5 above)
Comment (tromai) [reply docs/source/pages/cli_usage/action_dump_defaults.rst]: Addressed in https://github.com/oracle-samples/macaron/pull/269/commits/602d68376381eb591484d8df2ecaee0e26dd0e7e

File: docs/source/pages/cli_usage/action_analyze.rst (hunk 4 above)
Comment (tromai) [reply docs/source/pages/cli_usage/action_analyze.rst]: Addressed in https://github.com/oracle-samples/macaron/pull/269/commits/602d68376381eb591484d8df2ecaee0e26dd0e7e

File: docs/source/pages/cli_usage/action_analyze.rst (hunk 3 above)
Comment (tromai) [reply docs/source/pages/cli_usage/action_analyze.rst]: Addressed in https://github.com/oracle-samples/macaron/pull/269/commits/602d68376381eb591484d8df2ecaee0e26dd0e7e

**Review by behnazh-w — APPROVED** [review]

**Pull request merged**